packages/strapi-plugin-graphql/services/type-definitions.js

```diff
@@ -99,7 +99,7 @@ const buildTypeDefObj = model => {
 
 const generateEnumDefinitions = (attributes, globalId) => {
   return Object.keys(attributes)
-    .filter(attribute => attributes[attribute].type === 'enumeration')
+    .filter(attribute => attributes[attribute].type === 'enumeration' && !attributes[attribute].private)
```

Comment:
You should use `!isPrivateAttribute` instead

Reply:
This function needs a model as a parameter to be able to check if an attribute is private and the generateEnumbDefinitions does not have the model passed as a parameter that is why i am doing the check in the function itself.

Reply:
Pass the model instead of the attributes in the function instead. This way we make sure we can change the implementation if isPrivateAttributes & it will propagte at once

Reply:
Ok got you changing it.

Reply:
fixed